src/community/colormap/src/main/java/org/geotools/process/raster/FilterFunction_svgColorMap.java

```diff
@@ -27,15 +42,28 @@
  */
 public class FilterFunction_svgColorMap extends FunctionExpressionImpl {
     
-    public static final Color TRANSPARENT_COLOR = new Color(0,0,0,0); 
+    /**
+     * Two colors are used for before and after palette, so this leaves a max of 254 colors to play with
+     */
+    public static final int MAX_PALETTE_COLORS = 254;
+
+    private static final Logger LOGGER = Logging.getLogger(FilterFunction_svgColorMap.class);
+    
+    public static final Color TRANSPARENT_COLOR = new Color(0,0,0,0);
+    
+    private static StyleFactory SF = CommonFactoryFinder.getStyleFactory();
+    private static FilterFactory FF = CommonFactoryFinder.getFilterFactory();
     
     public static FunctionName NAME = new FunctionNameImpl("colormap",
             parameter("colormap", ColorMap.class),
             parameter("name", String.class),
             parameter("min", Number.class),
             parameter("max", Number.class),
             parameter("beforeColor", String.class, 0, 1),
-            parameter("afterColor", String.class, 0, 1));
+            parameter("afterColor", String.class, 0, 1),
+            parameter("afterColor", String.class, 0, 1),
+            parameter("logarithmic", Boolean.class, 0, 1),
```

Comment:
is afterColor param being added 2 times (line 63 and line 64) on purpose or is it simply a copy and paste TYPO?